### DISCUSSION TITLE
Create issue templates

### DISCUSSION THREAD

**Discussion by madsnedergaard** [electricitymaps/electricitymaps-contrib#3317] [Ideas] [answered]
It would be nice to have a few issue templates here :)

And now we can even use proper inputs to make the user experience better (e.g. not requiring contributors to know markdown): https://dev.to/github/how-to-build-google-like-forms-with-github-3ig2

This is open for anyone, I'd suggest taking a look at some of the existing issues to see what makes sense. Perhaps one for "Add [country]" updates/questions could make sense? Or maybe a "[] zone is down" type of issues?

**Comment by clawfire** [comment, marked as answer]
I would gladely help with that.

I played a lot with issue template and issue forms on my OSS project. But keep in mind people will still be able to open a ticket and write random stuff in it 😁

**Reply by madsnedergaard** [reply]
That would be awesome! 💪 
And yes, there will still be some random/rogue issues, but I hope the majority of the issues would benefit from it and somewhat follow the schema. 

Feel free to take a look at what type of issues have been coming in and how to "group"/theme them into a few common types - and let us know if you need help on this, some of the long-time contributors would also have great input on this! :)

**Reply by clawfire** [reply]
So far I see those patterns: 

1. BUGS
2. FEATURES
3. DATA

**BUGS**

- Bug on the website
- Bug on the app

Why 2 templates ? Well there's a lot you can ask from the user about their environment so ... better split the questions in 2 forms

**DATA**

- Data issues : reporting some bugs in the data which are not spotted by the bot(s)
- Updated sources of data : report updated source of data if you cannot / don't want to work on it yourself
- New production source request: ask for support a new production source. 

**FEATURES**

- Asking for a new feature

**Reply by madsnedergaard** [reply]
This is great! 🙏 

A few comments:
- For now I think it's okay grouping the two bug category into one and making it possible to select which platform (web/mobile) it's about. I think there's so few of these that we can just have a "Other relevant info (browser, OS, etc.)" section for env-details instead.
- For data I think the `data issues` is really good and probably the most used one!
- Could we consider merging the two other data use cases into one called something like "New data available", where a user can select if it's capacity data, real-time production data, prices, etc.? I think that would make things simpler :)
- Having one for features makes great sense, although we've been struggling with whether to use Github discussions for these type of requests instead. Have you had any experience with this? Would be great if we could somehow redirect people to the discussions page instead, but I don't know how 😄

**Reply by clawfire** [reply]
In order 😁 :

* We can start like this absolutely. Can provide a link to a site where they can copy paste the browser / devices infos like https://faisalman.github.io/ua-parser-js/#demo
* Can merge the data source into one
* Well, before discussion I was using the feature request issue as a place to discuss but it comes quickly a mess and not something very usefull if issues are "cards" you work with for your dev team I agree. Although, even if we cannot prevent someone to fill an issue we CAN display a warning and a link to the discussion 😃 (just no plain redirects)

**Reply by madsnedergaard** [reply]
That sounds great, thanks for the feedback! Feel free to create a PR with these when/if you have the time 😎
